src/app/features/course/courses-section/courses-section.component.ts

```diff
@@ -1,16 +1,199 @@
-import {Component, OnInit} from '@angular/core';
+import {
+  Component,
+  OnChanges,
+  OnInit,
+  DoCheck,
+  AfterContentInit,
+  AfterContentChecked,
+  AfterViewInit,
+  AfterViewChecked,
+  OnDestroy
+} from '@angular/core';
+import {Course} from '../../../core/models/course';
 
 @Component({
   selector: 'app-section',
   templateUrl: './courses-section.component.html',
   styleUrls: ['./courses-section.component.scss']
 })
-export class CoursesSectionComponent implements OnInit {
+export class CoursesSectionComponent implements OnChanges,
+                                                OnInit,
+                                                DoCheck,
+                                                AfterContentInit,
+                                                AfterContentChecked,
```

Comment:
I think is redundant to implement all these dependencies without practical usage

Reply:
all the dependencies were implemented due to this point:
![image](https://user-images.githubusercontent.com/12123267/67522458-71301200-f6b5-11e9-8cc5-078db22f5d9e.png)
will be removed.